app/vmagent/remotewrite/remotewrite.go

```diff
@@ -630,20 +630,42 @@ func (rwctx *remoteWriteCtx) Push(tss []prompbmarshal.TimeSeries) {
 	rowsCount := getRowsCount(tss)
 	rwctx.rowsPushedAfterRelabel.Add(rowsCount)
 
-	// Apply stream aggregation if any
+	releaseTss := func() {
+		// Return back relabeling contexts to the pool
+		if rctx != nil {
+			*v = prompbmarshal.ResetTimeSeries(tss)
+			tssPool.Put(v)
+			putRelabelCtx(rctx)
+		}
+	}
+	defer releaseTss()
```

Comment:
```suggestion
	defer func() {
		// Return back relabeling contexts to the pool
		if rctx != nil {
			*v = prompbmarshal.ResetTimeSeries(tss)
			tssPool.Put(v)
			putRelabelCtx(rctx)
		}
	}
```

defer can be called in relabeling clause, so obj is release only if it was used